XenAdmin/Commands/DisableChangedBlockTrackingCommand.cs

```diff
@@ -0,0 +1,150 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using XenAdmin.Actions;
+using XenAdmin.Core;
+using XenAPI;
+
+
+namespace XenAdmin.Commands
+{
+    /// <summary>
+    /// Disable changed block tracking of the selected VM.
+    /// </summary>
+    internal class DisableChangedBlockTrackingCommand : Command
+    {
+        /// <summary>
+        /// Initializes a new instance of this Command. The parameter-less constructor is required if 
+        /// this Command is to be attached to a ToolStrip menu item or button. It should not be used in any other scenario.
+        /// </summary>
+        public DisableChangedBlockTrackingCommand()
+        {
+        }
+
+        public DisableChangedBlockTrackingCommand(IMainWindow mainWindow, IEnumerable<SelectedItem> selection)
+            : base(mainWindow, selection)
+        {
+        }
+
+        public DisableChangedBlockTrackingCommand(IMainWindow mainWindow, IXenObject xenObject)
+            : base(mainWindow, xenObject)
+        {
+        }
+
+        private void Execute(IList<VM> vms)
+        {
+            var actions = new List<AsyncAction>();
+
+            foreach (var vm in vms)
+            {
+                if (vm.is_a_template)
+                    continue;
+                foreach (VBD vbd in vm.Connection.ResolveAll(vm.VBDs))
+                {
+                    VDI vdi = vm.Connection.Resolve(vbd.VDI);
+                    if (vdi != null && vdi.cbt_enabled)
+                        actions.Add(new VDIDisableCbtAction(vm, vdi));
+                }
+            }
+
+            if (actions.Any())
+            {
+                if (actions.Count == 1)
+                {
+                    actions[0].RunAsync();
+                }
+                else
+                {
+                    new ParallelAction(
+                        Messages.ACTION_DISABLE_CHANGED_BLOCK_TRACKING,
+                        Messages.ACTION_DISABLING_CHANGED_BLOCK_TRACKING,
+                        Messages.ACTION_DISABLED_CHANGED_BLOCK_TRACKING,
+                        actions).RunAsync();
+                }
+            }
+        }
+
+        protected override void ExecuteCore(SelectedItemCollection selection)
+        {
+            Execute(selection.AsXenObjects<VM>());
+        }
+
+        private bool CanExecute(VM vm)
+        {
+            return vm != null &&
+                !vm.is_a_template &&
+                vm.Connection.ResolveAll(vm.VBDs).Any(vbd => vm.Connection.Resolve(vbd.VDI) != null && vm.Connection.Resolve(vbd.VDI).cbt_enabled);
+        }
+
+        protected override bool CanExecuteCore(SelectedItemCollection selection)
+        {
+            // Can execute criteria: A selection of VMs in the same pool which is Inverness+, where at least one VM having CBT enabled
+            if (selection.Any() &&

```

Comment:
`Selection.Any()` is not needed here, because  `selection.AllItemsAre<VM>()` will return false is selection is empty.
So this could be written as
```
return selection.AllItemsAre<VM>() && selection.GetConnectionOfAllItems() != null &&
    Helpers.InvernessOrGreater(selection.GetConnectionOfFirstItem()) && selection.AtLeastOneXenObjectCan<VM>(CanExecute);
```